plugins/search/src/components/SearchBar/SearchBar.tsx

```diff
@@ -14,14 +14,61 @@
  * limitations under the License.
  */
 
-import React, { ChangeEvent, useEffect, useState } from 'react';
+import React, { ChangeEvent, useEffect, KeyboardEvent, useState } from 'react';
 import { useDebounce } from 'react-use';
 import { InputBase, InputAdornment, IconButton } from '@material-ui/core';
 import SearchIcon from '@material-ui/icons/Search';
 import ClearButton from '@material-ui/icons/Clear';
 
 import { useSearch } from '../SearchContext';
 
+type PresenterProps = {
+  value: string;
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  handleClear?: () => void;
+  onKeyDown?: (e: KeyboardEvent<HTMLInputElement>) => void;
+  className?: string;
+  placeholder?: string;
+};
+
+export const SearchBarPresenter = ({
+  value,
+  onChange,
+  onKeyDown,
+  handleClear,
+  className,
+  placeholder,
+}: PresenterProps) => {
+  const endAdornment = (
+    <InputAdornment position="end">
+      <IconButton aria-label="Clear term" onClick={handleClear}>
+        <ClearButton />
+      </IconButton>
+    </InputAdornment>
+  );
+
+  return (
+    <InputBase
+      data-testid="search-bar-next"
+      fullWidth
+      placeholder={placeholder ?? 'Search in Backstage'}
+      value={value}
+      onChange={onChange}
+      inputProps={{ 'aria-label': 'Search term' }}
+      startAdornment={
+        <InputAdornment position="start">
+          <IconButton aria-label="Query term" disabled>
+            <SearchIcon />
+          </IconButton>
+        </InputAdornment>
+      }
+      {...(handleClear ? { endAdornment } : {})}
```

Comment:
Hm. How about always showing this (it's a great end user use case anyway), and doing `onChange('')` when it's clicked? And skip the explicit clear callback in the first place

Reply:
That would make for a nicer API. However, as it is right now, `onChange` expects the actual event. 
In the same vein as in a previous comment - removing this makes the API less powerful, as it loses the access to the event. However, right now I can't really see a use-case for this other than getting the value. Getting free support for the Clear is probably worth the trade-off.

Fixed in c0a1398.

Reply:
That would make for a nicer API. However, as it is right now, `onChange` expects the actual event. 
In the same vein as in a previous comment - removing this makes the API less powerful, as it loses the access to the event. However, right now I can't really see a use-case for this other than getting the value. Getting free support for the Clear is probably worth the trade-off.

Fixed in c0a1398.